### PR TITLE
Restores 'add to' and 'save' output window features

### PR DESCRIPTION
**Pull request by Patowhiz**
Fixes partly #8408

@africanmathsinitiative/developers this is ready for review.
This is a temporary solution. It restore previously regressed features in a way that it was previously implemented.
Html outputs will still not be saved.

### CODE REVIEW

**Comment by rdstern** [comment]
@Patowhiz looks very promising already.  I tried with one variable summarise and also one variable graph.

It saved the graph nicely, but not the summary - there it seems to save a link to the file instead.

On details it doesn't yet seem to remember the directory I save it to.  It always seemed to come back to the Documents directory.

In the future I am hoping the Save in the Output window might vbe a Save - rather than Save As?  The only Save As will be from the File menu? 

But I'll take anything working for the start - as soon as it saves the txt output, rather than a link we have the basics.

**Comment by Patowhiz** [comment]
@rdstern I have fixed the text output and remembering of directory. 
Saving as links is definitely not desirable, considering that a user may want to share the file to another user with a different machine.

@rdstern I'm not sure if I should add links for htmls for now? Note this feature was there in the older versions anyway. 
To correctly implement html saving, I need some time.

**Review by rdstern — APPROVED** [review]
@Patowhiz that's a relief that it now works.  
As you mention, there's lots to improve still, including html, but I'd be very happy if this could now be merged.  So, over to @lloyddewit now.
I then hope that the work does not stop, so the output maximised windows have more possibilities in the mid-August release.

File: instat/UserControl/ucrOutputPage.vb [hunk 1]
```diff
@@ -403,10 +418,17 @@ Public Class ucrOutputPage
...
             Case OutputType.Script
-                FillRichTextBoxWithFormatedRScript(richText, element.FormattedRScript)
+                FillRichTextWithRScriptBasedOnSettings(richText, element.FormattedRScript)
             Case OutputType.TextOutput
-                'todo. check if output is file or not. if file, read the contents of the file
-                AddFormatedTextToRichTextBox(richText, element.Output, OutputFont.ROutputFont, OutputFont.ROutputColour)
+                Dim strOutput As String = ""
+                If element.IsFile Then
+                    For Each strLine As String In IO.File.ReadLines(element.Output)
+                        strOutput = strOutput & strLine & Environment.NewLine
```
Comment (lloyddewit) [inline instat/UserControl/ucrOutputPage.vb:426]:
```suggestion
                        strOutput &= strLine & Environment.NewLine
```

File: instat/Model/Output/clsOutputLogger.vb [hunk 2]
```diff
@@ -77,36 +77,47 @@ Public Class clsOutputLogger
...
+                                    "Developer Error",
+                                    MessageBoxButtons.OK,
+                                    MessageBoxIcon.Error)
+                        Exit Sub
+                End Select
+            Else
+                outputType = OutputType.TextOutput
+            End If
+
+            'add the output with it's R script as another output element
```
Comment (lloyddewit) [inline instat/Model/Output/clsOutputLogger.vb:111]:
```suggestion
            'add the output with its R script as another output element
```

**Pull request merged**